starknet_py/net/full_node_client.py

```diff
@@ -101,6 +104,75 @@ async def get_block_traces(
+        """.
+        :param from_block_number: Number of the block from which events searched for **start**
+                                or literals `"pending"` or `"latest"`.
+                                Mutually exclusive with ``from_block_hash`` parameter.
+        :param from_block_hash: Hash of the block from which events searched for **start**
+                                or literals `"pending"` or `"latest"`.
+                                Mutually exclusive with ``from_block_number`` parameter.
+        :param to_block_number: Number of the block from which events searched for **end**
+                                or literals `"pending"` or `"latest"`.
+                                Mutually exclusive with ``to_block_hash`` parameter.
+        :param to_block_hash: Hash of the block from which events searched for **end**
+                                or literals `"pending"` or `"latest"`.
+                                Mutually exclusive with ``to_block_number`` parameter.
+        :param address: The address of the contract that emitted the event.
+        :param keys: List of names of events that are searched for.
+        :param follow_continuation_token: Flag deciding whether all events should be collected during one function call,
+                                defaults to False.
+        :param chunk_size: Size of chunk of events returned by one ``get_events`` call, defaults to 1 (minimum).
+
+        :returns: ``EventsResponse`` dataclass containing events and optional continuation token.
+        """
+        params = {"chunk_size": chunk_size if chunk_size >= 1 else 1,
+                  "from_block": get_block_identifier_for_get_events(from_block_hash, from_block_number),
+                  "to_block": get_block_identifier_for_get_events(to_block_hash, to_block_number),
+                  "address": _to_rpc_felt(address)}
+
+        keys = [_starknet_keccak(bytes(i, "utf-8")) for i in keys]
+        params["keys"] = list(map(_to_rpc_felt, keys))
+        previous_continuation_token = None
+        while True:
+            events_response = await self.get_events_chunk(params)
+            continuation_token = events_response.continuation_token
+            # TODO fix the condition after devnet change, should be `or not continuation_token`.
+            # As of now, devnet returns previous continuation token when there are no events.
+            # However, the json "result" part should only contain `"events": []` and nothing else.
+            # Right now we must check whether the last returned continuation token is the same as the previous one.
+            # If so, the loop should break.
+            if not follow_continuation_token or continuation_token == previous_continuation_token:
+                break
+            params["continuation_token"] = continuation_token
+            previous_continuation_token = continuation_token
+
+        return events_response
```

Comment:
Shouldn't the events somehow be aggregated? Will all events be returned if `follow_continuation_token` is True?